packages/data-context/src/actions/EventCollectorActions.ts

```diff
@@ -0,0 +1,34 @@
+import type { DataContext } from '..'
+import Debug from 'debug'
+
+const debug = Debug('cypress:data-context:sources:EventCollectorDataSource')
+
+interface CollectableEvent {
+  campaign: string
+  messageId: string
+  medium: string
+}
+
+export class EventCollectorActions {
+  constructor (private ctx: DataContext) {}
+
+  async recordEvent (event: CollectableEvent): Promise<boolean> {
+    try {
+      // TODO: Stubbed with Staging for development
+      const dashboardUrl = 'https://dashboard-staging.cypress.io' //this.ctx.cloud.getDashboardUrl()
```

Comment:
Should we have some kind of process.env check here, or will this change in a subsequent PR?

Reply:
Edit I saw 

> Endpoint is currently hardcoded to use Staging so we do not unnecessarily pollute Prod data during development - we can filter this out if needed but no sense recording to Prod before we have complete data


Disregard this

Reply:
One thing to note here - the runner is now gathering data from production cloud by default when in development. So `this.ctx.cloud.getDashboardUrl()` will resolve to the production cloud URL because `DEFAULT_INTERNAL_CLOUD_ENV` will be `production`.

We probably don't want to send these events to production when in dev mode, so we might want to use `CYPRESS_INTERNAL_ENV` instead, and only use the production URL when that variable is `production`.

Reply:
@astone123 Good point. I'd convinced myself they should be the same but you're probably right, we don't want to write events to production when running locally even though we're reading data from prod